src/loaders/ObjectLoader.js

```diff
@@ -511,15 +280,21 @@ class ObjectLoader extends Loader {
 
 					case 'Geometry':
 
-						console.error( 'THREE.ObjectLoader: Loading "Geometry" is not supported anymore.' );
+						console.error( 'THREE.ObjectLoader: The legacy Geometry type is no longer supported.' );
 
 						break;
 
 					default:
 
-						console.warn( 'THREE.ObjectLoader: Unsupported geometry type "' + data.type + '"' );
+						if ( data.type in Geometries ) {
+
+							geometry = Geometries[ data.type ].fromJSON( data, shapes );
```

Comment:
Hmm, this is not the prettiest...
Only `ExtrudeGeometry` and `ShapeGeometry` uses the `shapes` parameter 🤔

Reply:
We could consider to resolve the shape UUIDs in `ObjectLoader` and then store the resulting shapes in the `data` object. In this way, all `fromJSON()` method would have a common signature. 

However, it would be necessary to check for `data.shapes` in `ObjectLoader` for each geometry.

Reply:
Yeah, I'm not sure what's better...
I think I'll try the current approach for now.